src/main/java/org/akip/service/TaskInstanceService.java

```diff
@@ -151,7 +151,7 @@ private void checkCurrentUserPermission(List<String> candidateGroups) {
             }
         }
 
-        throw new ClaimNotAllowedException(String.join(", ", candidateGroups));
+        throw new BadRequestErrorException("Task reserved for users " + String.join(", ", candidateGroups));
```

Comment:
I was wondering what happens whether the task has no candidate groups but an assignee instead.
Did you try this scenario? I think you should improve the message or forget about giving this information about who can execute the task.